src/fast_check/transform_dts.rs

```diff
@@ -0,0 +1,730 @@
+    Self { id_counter: 0 }
+  }
+
+  fn gen_unique_name(&mut self) -> String {
+    self.id_counter += 1;
+    format!("_dts_{}", self.id_counter)
+  }
+
+  pub fn transform(
+    &mut self,
+    mut module: Module,
+  ) -> Result<Module, Vec<FastCheckDiagnostic>> {
+    let mut body = module.body;
+
+    let mut new_items: Vec<ModuleItem> = vec![];
+
+    for item in &mut body {
+      match item {
+        ModuleItem::ModuleDecl(module_decl) => match module_decl {
+          ModuleDecl::Import(_) => {
+            new_items.push(item.clone());
+          }
+          ModuleDecl::ExportDecl(export_decl) => {
+            if let Some(decl) = self.decl_to_type_decl(&export_decl.decl) {
+              new_items.push(ModuleItem::ModuleDecl(ModuleDecl::ExportDecl(
+                ExportDecl {
+                  decl,
+                  span: DUMMY_SP,
+                },
+              )));
+            }
+          }
+          ModuleDecl::ExportDefaultDecl(export_decl) => {
+            match &mut export_decl.decl {
+              DefaultDecl::Class(class_expr) => {
+                class_expr.class.body =
+                  self.class_body_to_type(&class_expr.class.body);
+                new_items.push(item.clone());
+              }
+              DefaultDecl::Fn(fn_expr) => {
+                fn_expr.function.body = None;
+                new_items.push(item.clone());
+              }
+              DefaultDecl::TsInterfaceDecl(_) => new_items.push(item.clone()),
+            }
+          }
+          ModuleDecl::ExportDefaultExpr(export_default_expr) => {
+            let name = self.gen_unique_name();
+            let name_ident = Ident::new(name.into(), DUMMY_SP);
+            let type_ann = self
+              .expr_to_ts_type(*export_default_expr.expr.clone(), false)
+              .map(type_ann)
+              .or(Some(any_type_ann()));
+
+            new_items.push(ModuleItem::Stmt(Stmt::Decl(Decl::Var(Box::new(
+              VarDecl {
+                span: DUMMY_SP,
+                kind: VarDeclKind::Const,
+                declare: true,
+                decls: vec![VarDeclarator {
+                  span: DUMMY_SP,
+                  name: Pat::Ident(BindingIdent {
+                    id: name_ident.clone(),
+                    type_ann,
+                  }),
+                  init: None,
+                  definite: false,
+                }],
+              },
+            )))));
+
+            new_items.push(ModuleItem::ModuleDecl(
+              ModuleDecl::ExportDefaultExpr(ExportDefaultExpr {
+                span: DUMMY_SP,
+                expr: Box::new(Expr::Ident(name_ident)),
+              }),
+            ))
+          }
+          // Keep all these
+          ModuleDecl::TsImportEquals(_)
+          | ModuleDecl::TsNamespaceExport(_)
+          | ModuleDecl::TsExportAssignment(_) => {
+            new_items.push(ModuleItem::ModuleDecl(module_decl.clone()));
+          }
+          _ => {}
```

Comment:
Instead of doing a wildcard match (`_ =>`), we can change this to list all `ModuleDecl` variants that will match for this. That way when new a new `ModuleDecl` variant gets added we can get errors here and decide how we should update the code.

Similarly we should do this for all wildcards below. Also, we should probably surface an error for these scenarios (to say we had issues creating a dts file) or debug panic.